api4/channel.go

```diff
@@ -179,6 +180,37 @@ func patchChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func restoreChannel(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	var channel *model.Channel
+	var err *model.AppError
+	if channel, err = app.GetChannel(c.Params.ChannelId); err != nil {
+		c.Err = err
+		return
+	}
+	teamId := channel.TeamId
+
+	if !app.SessionHasPermissionToTeam(c.Session, teamId, model.PERMISSION_MANAGE_TEAM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_TEAM)
+		return
+	}
+
+	channel, err = app.RestoreChannel(channel)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	c.LogAudit("name=" + channel.Name)
+	w.WriteHeader(http.StatusOK)
```

Comment:
You don't actually need to write the 200 header back, it does that automatically if no status code is provided